.github/workflows/docs.yml

```diff
@@ -0,0 +1,59 @@
+name: Generate documentation
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main", "publish-api-reference"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+
+jobs:
+  ci:
+    name: Build Doxygen documentation
+
+    runs-on: macos-14
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Dependencies
+        uses: ./.github/actions/setup-dependencies
+        with:
+          use_ccache: false
+
+      - name: Install awscli
+        run: brew install awscli || true
+
+      - name: Install doxygen and graphviz (a dependency of Doxygen for generating diagrams)
+        run: brew install doxygen graphviz || true
+
+      - name: Generate Doxygen Documentation for Slice
+        working-directory: ./doxygen
+        run: doxygen
+
+      - name: Generate Doxygen Documentation for C++
+        working-directory: ./cpp
+        run: |
+          make generate-srcs
+          cd doxygen
+          doxygen
+
+      # This will perform a full sync of the documentation to S3 every time the workflow is run since
+      # the timestamps will always be different. Using --size-only is not sufficient since the
+      # documentation may be updated without changing the size of the files. S3 does not offer a hash based sync.
+      #
+      # Additionally, we do not cache the doxygen output since it does not remove files old files.
+      - name: Sync Documentation to S3
```

Comment:
why are we publishing the docs for main on S3?

Reply:
The plan is it to have have the docs public for the for various branches. No need for our internal site.

We you can always see the latest api ref at:

https://doc.zeroc.com/api/ice/main/cpp/index.html